common/components/ui/Modal.jsx

```diff
@@ -0,0 +1,107 @@
+// @flow
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import closeIcon from 'assets/images/icon-x.svg';
+
+import './Modal.scss';
+
+export default class Modal extends Component {
+  static propTypes = {
+    isOpen: PropTypes.bool,
```

Comment:
This was my first instinct as well, but after looking at [react-modal](https://github.com/reactjs/react-modal), they did this to allow for internally managed animations. I didn't have any fancy close animations going on for this one, but I wanted to choose the most flexible option for future changes to the modals.

If you feel like this isn't a worthwhile consideration, I'd be happy to ditch it.